packages/ckeditor5-engine/src/model/nodelist.ts

```diff
@@ -194,7 +195,7 @@ export default class NodeList implements Iterable<Node> {
 			}
 		}
 
-		this._nodes.splice( index, 0, ...nodes );
+		spliceArray<Node>( this._nodes, Array.from( nodes ), index, 0 );
```

Comment:
Although it is not needed by current implementation (as the function "works in place"), I'd go with `this._nodes = spliceArray...` to be future-proof.

Reply:
Same thing in other places in this PR.

Reply:
:thinking: I am not sure why it is readonly, it isn't like that in original .js file. @arkflpc

Reply:
Private fields are not in the contract of the class. You can drop `readonly` modifier anytime.